qiskit_nature/problems/second_quantization/vibrational/vibrational_problem.py

```diff
@@ -0,0 +1,56 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""The Vibrational Problem class."""
+from typing import List, Tuple, Optional
+
+from qiskit_nature import WatsonHamiltonian
+from qiskit_nature.drivers import BosonicDriver
+from qiskit_nature.operators.second_quantization import SecondQuantizedOp
+from qiskit_nature.problems.second_quantization.base_problem import BaseProblem
+from qiskit_nature.problems.second_quantization.vibrational.spin_op_builder import build_spin_op
+from qiskit_nature.transformers import BaseTransformer
+
+
+class VibrationalProblem(BaseProblem):
+    """Vibrational Problem"""
+
+    def __init__(self, bosonic_driver: BosonicDriver,
+                 transformers: Optional[List[BaseTransformer]] = None):
+        """
+
+        Args:
+            bosonic_driver: A bosonic driver encoding the molecule information.
+            transformers: A list of transformations to be applied to the molecule.
+        """
+        super().__init__(bosonic_driver, transformers)
+
+    def second_q_ops(self) -> List[SecondQuantizedOp]:
+        """Returns a list of `SecondQuantizedOp` created based on a driver and transformations
+        provided.
+
+        Returns:
+            A list of `SecondQuantizedOp` in the following order: ... .
+        """
+        watson_hamiltonian = self.driver.run()
+        watson_hamiltonian_transformed = self._transform(watson_hamiltonian)
+        basis_size = 1  # TODO how to get it?
+        truncation_order = 3  # TODO how to get it?
```

Comment:
Afaict from the old way of using `BosonicTransformation` (https://github.com/Qiskit/qiskit-aqua/blob/master/test/chemistry/test_bosonic_esc_calculation.py#L60-L64) this will need to be an input to the constructor. @paulineollitrault Please correct me, if these can be inferred in some other way.

Reply:
I have been inclined towards putting it to the `VibrationalProblem` constructor. Another option would be arguments of `second_q_ops` but it conflicts with a signature of that method in a parent class.

Reply:
No `second_q_ops` should not have any arguments. Once you call that, everything should really be set.
The idea here is that it should feel like a property that you query